collector-http/src/main/java/zipkin2/collector/otel/http/OpenTelemetryHttpCollector.java

```diff
@@ -52,38 +77,87 @@ public static final class Builder extends CollectorComponent.Builder {
   }
 
   final Collector collector;
+
   final CollectorMetrics metrics;
 
   OpenTelemetryHttpCollector(Builder builder) {
     collector = builder.delegate.build();
     metrics = builder.metrics;
   }
 
-  @Override public OpenTelemetryHttpCollector start() {
+  @Override
+  public OpenTelemetryHttpCollector start() {
     return this;
   }
 
-  @Override public String toString() {
+  @Override
+  public String toString() {
     return "OpenTelemetryHttpCollector{}";
   }
 
   /**
    * Reconfigures the service per https://opentelemetry.io/docs/specs/otlp/#otlphttp-request
    */
-  @Override public void reconfigure(ServerBuilder sb) {
+  @Override
+  public void reconfigure(ServerBuilder sb) {
+    sb.decorator(DecodingService.newDecorator(StreamDecoderFactory.gzip()));
     sb.service("/v1/traces", new HttpService(this));
   }
 
   static final class HttpService extends AbstractHttpService {
+    private static final Logger LOG = Logger.getLogger(HttpService.class.getName());
+
     final OpenTelemetryHttpCollector collector;
 
     HttpService(OpenTelemetryHttpCollector collector) {
       this.collector = collector;
     }
 
-    @Override protected HttpResponse doPost(ServiceRequestContext ctx, HttpRequest req)
-        throws Exception {
-      throw new RuntimeException("Implement me!");
+    @Override
+    protected HttpResponse doPost(ServiceRequestContext ctx, HttpRequest req) {
+      CompletableCallback result = new CompletableCallback();
+      req.aggregate(AggregationOptions.usePooledObjects(ctx.alloc(), ctx.eventLoop()
+      )).handle((msg, t) -> {
+        if (t != null) {
+          result.onError(t);
+          return null;
+        }
+        try (HttpData content = msg.content()) {
+          if (content.isEmpty()) {
+            result.onSuccess(null);
+            return null;
+          }
+
+          try {
+            ExportTraceServiceRequest request = ExportTraceServiceRequest.parseFrom(UnsafeByteOperations.unsafeWrap(content.byteBuf().nioBuffer()).newCodedInput());
+            List<Span> spans = SpanTranslator.translate(request);
+            collector.collector.accept(spans, result);
+          }
+          catch (IOException e) {
+            LOG.log(Level.WARNING, "Unable to parse the request:", e);
+            result.onError(new UncheckedIOException(e));
```

Comment:
if not rethrow, just use `result.onError(e);` is better?

Reply:
agree. fixed in https://github.com/openzipkin-contrib/zipkin-otel/pull/13/commits/b16231134eaa672e14436a49f95f939043e08261